packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -0,0 +1,237 @@
+            throw new TypeError('Variable does not support contains')
+          }
+        })(${args[0]}, ${args[1]})`
+      case 'matches': return `((str, regex) => {
+          if (${isString('str')}) {
+            if (Object.getPrototypeOf(regex) === RegExp.prototype) {
+              return RegExp.prototype.test.call(regex, str)
+            } else if (${isString('regex')}) {
+              return String.prototype.match.call(str, regex) !== null
+            } else {
+              throw new TypeError('Regular expression must be either a string or an instance of RegExp')
+            }
+          } else {
+            throw new TypeError('Variable is not a string')
+          }
+        })(${args[0]}, ${args[1]})`
+      case 'filter': return `(($dd_var) => {
+          return ${isIterableCollection('$dd_var')}
+            ? Array.from($dd_var).filter(($dd_it) => ${args[1]})
+            : Object.entries($dd_var).filter(([$dd_key, $dd_value]) => ${args[1]}).reduce((acc, [k, v]) => {
+                acc[k] = v
+                return acc
+              }, {})
+        })(${args[0]})`
+      case 'substring': return `((str) => {
+          if (${isString('str')}) {
+            return String.prototype.substring.call(str, ${args[1]}, ${args[2]})
+          } else {
+            throw new TypeError('Variable is not a string')
+          }
+        })(${args[0]})`
+      case 'getmember': return accessProperty(args[0], args[1], false)
+      case 'index': return accessProperty(args[0], args[1], true)
+    }
+  }
+
+  throw new TypeError(`Unknown AST node type: ${type}`)
+}
+
+function iterateOn (fnName, variable, callbackCode) {
+  return `(($dd_val) => {
+    return ${isIterableCollection('$dd_val')}
+      ? Array.from($dd_val).${fnName}(($dd_it) => ${callbackCode})
+      : Object.entries($dd_val).${fnName}(([$dd_key, $dd_value]) => ${callbackCode})
+  })(${variable})`
+}
+
+function isString (variable) {
+  return `typeof ${variable} === 'string' || ${variable} instanceof String`
+}
+
+function isIterableCollection (variable) {
+  return `${isArrayOrTypedArray(variable)} || ${variable} instanceof Set || ${variable} instanceof WeakSet`
+}
+
+function isArrayOrTypedArray (variable) {
+  return `Array.isArray(${variable}) || ${variable} instanceof Object.getPrototypeOf(Int8Array)`
+}
+
+function getSize (variable) {
+  return `((val) => {
+    if (${isString('val')} || ${isArrayOrTypedArray('val')}) {
+      return ${guardAgainstPropertyAccessSideEffects('val', '"length"')}
+    } else if (val instanceof Set || val instanceof Map) {
+      return ${guardAgainstPropertyAccessSideEffects('val', '"size"')}
+    } else {
+      throw new TypeError('Cannot get length or size of string/collection')
+    }
+  })(${variable})`
+}
+
+function accessProperty (variable, keyOrIndex, allowMapAccess) {
+  return `((val, key) => {
+    if (val instanceof Set || val instanceof WeakSet) {
+      throw new Error('Accessing a Set or WeakSet is not allowed')
```

Comment:
Nit: I would always put these to the end. That way the positive ones are handled faster.